jdbc-v2/src/test/java/com/clickhouse/jdbc/internal/JdbcConfigurationTest.java

```diff
@@ -1,31 +1,102 @@
 package com.clickhouse.jdbc.internal;
 
-import com.clickhouse.jdbc.JdbcIntegrationTest;
-import org.testng.annotations.BeforeTest;
+import com.clickhouse.client.api.ClientConfigProperties;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.sql.DriverPropertyInfo;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
 import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import static org.testng.Assert.assertEquals;
 
-public class JdbcConfigurationTest extends JdbcIntegrationTest {
-    @Test(groups = { "integration" })
-    public void testCleanUrl() {
-        JdbcConfiguration configuration = new JdbcConfiguration("jdbc:clickhouse://localhost:8123/clickhouse?param1=value1&param2=value2", new Properties());
-
-        String url = "jdbc:clickhouse://localhost:8123/clickhouse?param1=value1&param2=value2";
-        String cleanUrl = configuration.cleanUrl(url);
-        assertEquals(cleanUrl, "http://localhost:8123/clickhouse?param1=value1&param2=value2");
-
-        // we do not support guessing by port numbers
-        url = "jdbc:clickhouse://localhost:8443/clickhouse?param1=value1&param2=value2";
-        cleanUrl = configuration.cleanUrl(url);
-        assertEquals(cleanUrl, "http://localhost:8443/clickhouse?param1=value1&param2=value2");
-
-        Properties info = new Properties();
-        configuration = new JdbcConfiguration("jdbc:clickhouse:https://localhost:8123/clickhouse?param1=value1&param2=value2", info);
-        url = "jdbc:clickhouse:https://localhost:8443/clickhouse?param1=value1&param2=value2";
-        cleanUrl = configuration.cleanUrl(url);
-        assertEquals(cleanUrl, "https://localhost:8443/clickhouse?param1=value1&param2=value2");
+public class JdbcConfigurationTest {
+
+    @Test(dataProvider = "testConnectionUrlDataProvider")
+    public void testConnectionUrl(String jdbcUrl, String connectionUrl, Properties properties, Map<String, String> expectedClientProps) throws Exception {
+        JdbcConfiguration configuration = new JdbcConfiguration(jdbcUrl, properties);
+        assertEquals(configuration.getConnectionUrl(), connectionUrl);
+        System.out.println(configuration.clientProperties);
```

Comment:
Do you want to convert into logger instead of System.out

Reply:
I will remove it.